Audit and fix theme token updates against KDS

MisRob opened the pull request:
## Summary

Updates Kolibri to KDS theming updates introduced by https://github.com/learningequality/kolibri-design-system/pull/551.

Contains KDS upgrade 3.0.1 -> 4.2.0. Release notes:
- [3.1.0](https://github.com/learningequality/kolibri-design-system/releases/tag/v3.1.0)
- [3.1.1](https://github.com/learningequality/kolibri-design-system/releases/tag/v3.1.1)
- [3.2.0](https://github.com/learningequality/kolibri-design-system/releases/tag/v3.2.0)
- [4.0.0](https://github.com/learningequality/kolibri-design-system/releases/tag/v4.0.0)
- [4.0.1](https://github.com/learningequality/kolibri-design-system/releases/tag/v4.0.1)
- [4.1.0](https://github.com/learningequality/kolibri-design-system/releases/tag/v4.1.0)
- [4.2.0](https://github.com/learningequality/kolibri-design-system/releases/tag/v4.2.0)

## References

- Closes https://github.com/learningequality/kolibri/issues/11867
- KDS PR: https://github.com/learningequality/kolibri-design-system/pull/551
- Designs: https://www.figma.com/file/p7aOP2MBv0SED1uhaPwqEd/Product-rebrand?type=design&node-id=426-7545&mode=design


## Reviewer guidance

- Please study the new theme first https://deploy-preview-551--kolibri-design-system.netlify.app/colors/#

### Development

- [ ] Are code changes fine?
- [ ] Read the upgrade process below and preview regexes I used. Can you think of any important area I left out?

### QA

- [ ] Can you see some glitches in regards to colors?
- [ ] Are contrast ratios good?
- [ ] Please test states that are often not obviously visible - errors, warning, and similar
- [ ] Mobile device

You may get a good sense on what to look from the "Upgrade process" notes below.

## Upgrade process

Available colors and tokens can be found at https://deploy-preview-551--kolibri-design-system.netlify.app/

### (1) Breaking KDS API changes

**(a) KDS removed palette colors: `purple`, `deeppurple`, `indigo`, `blue`, `brown`, `cyan`, `teal`, `lightgreen`, `lime`, `yellow`, `amber`, `deeporange`, `bluegrey`**

Search for

```bash
palette.purple|palette.deeppurple|palette.indigo|palette.brown|palette.cyan|palette.teal|palette.lightgreen|palette.lime|palette.amber|palette.deeporange|palette.bluegrey
```
and


```bash
palette\(\).purple|palette\(\).deeppurple|palette\(\).indigo|palette\(\).brown|palette\(\).cyan|palette\(\).teal|palette\(\).lightgreen|palette\(\).lime|palette\(\).amber|palette\(\).deeporange|palette\(\).bluegrey
```

and replace with closest available palette colors

**(b) KDS removed grey scale values: `palette.grey.v_300`, `palette.grey.v_500`, `palette.grey.v_700`, `palette.grey.v_900`**

Search for

```bash
palette.grey.v_300|palette.grey.v_500|palette.grey.v_700|palette.grey.v_900
```
and

```bash
palette\(\).grey.v_300|palette\(\).grey.v_500|palette\(\).grey.v_700|palette\(\).grey.v_900
```
and replace with the closest darker grey scale values that are available

**(c) KDS removed scale values for `brand` and `palette` colors (except `palette.grey`): `v_50`,`v_100`, `v_300`, `v_500`, `v_700`, `v_900`**

Search for

```bash
(?<!grey)\.v_(50|100|300|500|700|900)\b
```

and replace with the closest scale value that is available

**(d) KDS removed content-related tokens: `exercise`, `video`, `audio`, `document`, `html5`, `slideshow`**

Search for

```bash
tokens.exercise|tokens.video|tokens.audio|tokens.document|tokens.html5
```

and

```bash
tokens\(\).exercise|tokens\(\).video|tokens\(\).audio|tokens\(\).document|tokens\(\).html5
```

and replace with relevant tokens

**(e) KDS removed other tokens: `appBarFullscreen`, `appBarFullscreenDark`, `linkDark`**

Search for

```bash
tokens.appBarFullscreen|tokens.appBarFullscreenDark|tokens.linkDark
```

and

```bash
tokens\(\).appBarFullscreen|tokens\(\).appBarFullscreenDark|tokens\(\).linkDark
```

and replace with relevant tokens

### (2) Hardcoded color values

Search the most commonly used CSS properties that can receive color

```bash
color|background|border|outline-color|box-shadow|text-shadow|fill|stroke|outline|text-decoration-color
```

and locate any hex, rgb(a), hsl(a),named color values to be replaced with relevant colors (themed ideally, but if blocked replace with new hardcoded color and open follow-up)

### (3) Colors

Some colors in the new palette have the same name but look differently. Due to reduction in scales, they will now often look lighter. Preview colors and see if higher scale value is needed to ensure good contrast ratio or if another colors need to be used to work in the new theme. This applies particularly to non-grey colors.

```
(?<!grey)\.v_(200|400|600|800|1000|1100)\b
```

### (4) Pages background color

Previously, we used `palette.grey.v_100` to set page background color. It needs to be updated to lighter `.v_50`. There are more components where it's being set. Search for places that set some kind of background color with `palette.grey.v_100` and see if it needs to be changed to `palette.grey.v_50`.

### (5) Contrast

Check places that utilize `tokens.text` and `tokens.textInverted` together with setting background color via `palette` for sufficient contrast.

### (6) Brand colors

Check places where `brand.primary` and `brand.secondary` is used and see if they need to be switched.

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests


## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

rtibbles committed:
Rebrand: Update Kolibri loader to new branding.

MisRob committed:
Rebrand: Replace obsolete grey scales

In the new theme, palette.grey.v_300/500/700/900
are not available anymore. This updates all places
using removed scales by replacing them
with closest darker grey scales that are available.

MisRob committed:
Rebrand: Rebrand side navigation and menu option

MisRob committed:
Rebrand: Rebrand the app bar

MisRob committed:
Rebrand: Use lighter background for pages

MisRob committed:
Rebrand: Rebrand immersive toolbar

MisRob committed:
Rebrand: Lighten the nav bar link hover state

MisRob committed:
Rebrand: Improve hover contrast

MisRob committed:
Rebrand: Hardcode epub themes colors

Firstly, it'd be good to keep epub reader themes
indepdendent of Kolibri and its plugins theming
to prevent from unintended changes caused by theming.
Secondly, many these colors are now removed from
the new theme.

MisRob committed:
Rebrand: Replace content related tokens

MisRob committed:
Rebrand: Replace obsolete green scales

Replaces obsolete green scales with
scales availabe in the new theme.

The new green is much lighter than the
previous one so it needs to be replaced
by  much higher scale to achieve a similar
effect.

MisRob committed:
Rebrand: Replace obsolete red scales

Replaces obsolete red scales with
scales availabe in the new theme.

The new red is much lighter than the
previous one so it needs to be replaced
by  much higher scale to achieve a similar
effect.

MisRob committed:
Rebrand: Rebrand auth pages

MisRob committed:
Rebrand: Update brand scales in the side search

panel and category modal

MisRob commented on kolibri/core/assets/src/views/AppBar.vue:
```diff
@@ -12,7 +12,7 @@
         :removeNavIcon="isAppContext && isTouchDevice"
         type="clear"
-        textColor="white"
+        textColor="black"
```
Until we add full theming support to Keen UiToolbar, we need to use either 'white' or 'black'. This will be done later on as part of KDS maintenance.

MisRob committed:
Rebrand: Consistent immersive bar color on topics page

Guidelines in regards to black bar will be updated.
The black color should be used any time we show
back or close icon. This serves as a mode-switch
cue. This updates only one area. Other places
will be resolved in a follow-up.

MisRob committed:
Rebrand: Replace obsolete amber color

MisRob committed:
Rebrand: Use blue hover state in tabs

MisRob committed:
Rebrand: Use success token

MisRob committed:
Rebrand: Lighter bg color in Change facility

MisRob committed:
Rebrand: Update warning icons color

MisRob committed:
Rebrand: Use higher green scale for better contrast

in the import users preview table data.

MisRob committed:
Rebrand: Use success token

MisRob committed:
Rebrand: Update beginners chip color

MisRob committed:
Rebrand: Lighter background in the practice quiz

MisRob committed:
Rebrand: Blue hover state for side panel tabs

MisRob committed:
Install KDS v4.0.1

Release notes
- https://github.com/learningequality/kolibri-design-system/releases/tag/v3.1.0
- https://github.com/learningequality/kolibri-design-system/releases/tag/v4.0.0
- https://github.com/learningequality/kolibri-design-system/releases/tag/v4.0.1

AllanOXDi committed:
Rebrand: replaced hexa and rgb colors with theme palettes and tokens

akolson commented:
HI @MisRob! I update the regex `(?<!grey)\.v_(50|100|300|500|700|900)` to `(?<!grey)\.v_(50|100|300|500|700|900)\b` as it was picking up any v_* including the mentioned values and not the actual values. As such, I was able to pickup `v_1000`. I also updated a another similar to this.

MisRob commented:
Makes sense, thanks @akolson

AllanOXDi committed:
Rebrand: fixes typo

radinamatic commented:
After the discussion about the text input fields on Slack, and subsequent changes by @LianaHarris360, that component now seems to have sufficient contrast, but there is a similar contrast issue with `KSelect` in the Library page:

<img width="782" alt="KSelect" src="https://github.com/learningequality/kolibri/assets/1457929/46dbf776-0fe0-491e-9952-efd025345896">


@LianaHarris360 again dug into code and found out that `KSelect` component in the file `kolibri/plugins/learn/assets/src/views/SearchFiltersPanel/SelectGroup.vue` has the background manually set to `$themePalette.grey.v_200`  which is the HEX `#CCCCCC`, and that is way too dark for the labels to have sufficient contrast. 

Can we make sure to apply the same front- and background color combinations as for the input text fields? Thank you!

LianaHarris360 commented:
I also want to flag that the original `KSelect` component has no background color at all, but after taking another look at the Figma specs, it should have the same background color as the other input components. I will update the KDS PR to reflect this change.

MisRob commented:
Alright, thank you. I will remove select background color override from Kolibri.

MisRob committed:
Rebrand: Do not override select style

MisRob committed:
Rebrand: Update snapshots

MisRob commented:
Selects styling is fixed

MisRob commented:
Since there's still time, I will do some more work here fixing issues that @jtamiace reported https://www.notion.so/learningequality/Kolibri-rebranded-first-preview-a8c0dc1a63a24071931dac150af0c014

MisRob committed:
Rebrand: Use surface token for sign in box

MisRob committed:
Update composable paths

This is a part of upgrade to KDS v4.0.0
that changes location of the composable.

MisRob committed:
Rebrand: Fix grey scales

MisRob committed:
Rebrand: Use token for dividers color

MisRob committed:
Rebrand: Fix dividers color

MisRob committed:
Rebrand: Lighter thumbnail placeholder color

MisRob committed:
Rebrand: Make the button one scale darker

to pass contrast check.

MisRob committed:
Rebrand: Make selected text visible

MisRob committed:
Rebrand: Add hover color

MisRob committed:
Rebrand: Use token for dividers

MisRob committed:
Rebrand: Lighter bg color

MisRob committed:
Rebrand: List items bg color

MisRob committed:
Merge branch 'release-v0.16.x' into rebrand

MisRob committed:
Install the latest KDS

MisRob commented:
We still need to replace the logo, but other than that, I think everything that has come up so far is resolved - let's review and QA so there's time to incorporate new feedback.

pcenov commented:
Hi @MisRob I have identified a number of issues mainly about the behavior and presentation of the modals, the search fields and the drop-down lists. As I have too many screenshots and videos I have added them to this [shared folder](https://drive.google.com/drive/folders/1luIwDHwhIiTYqJoMKLwXZvWIGDFC7uiV) so that you can take a look and identify the ones that are caused by changes made by this PR.

LianaHarris360 commented:
Also, it appears that the `LanguageSwitcherList` component is not being displayed on the login page.

<img src="https://github.com/learningequality/kolibri/assets/46411498/65ae9641-9757-48b1-81de-05208daa73b4" width="500">

rtibbles commented:
Hrm, is it possible that we'll need to backport @AlexVelezLl's work from develop around this? There's this PR: https://github.com/learningequality/kolibri/pull/11977 but I also think some previous work as well.

AlexVelezLl commented:
Hmm Misha suggested not using the KListWithOverflow component in 0.16 because of IE support @rtibbles.

This seems to be a regression introduced in #11847, but I think we can just revert it if we can live with #7997 in 0.16.

And leave the more general solution from #11977 in develop.

rtibbles commented:
Does that mean that this issue is currently extant in 0.16 then?

AlexVelezLl commented:
Nvm its not from #11847. Just tested it:

![image](https://github.com/learningequality/kolibri/assets/51239030/3393db1d-8b0f-407b-9cae-a8fa29473b23)

rtibbles commented:
Well, that's a relief!

rtibbles commented:
Seems like it is a regression from this PR specifically then.

MisRob commented:
Thank you @LianaHarris360 and @pcenov for feedback, that's really valuable. I believe that after it's all resolved, the PR will be robust. @pcenov I think some of the issues will definitely be related to rebrand and some probably not. There's still time so I will see if I can work through it all here since it may be faster than opening so many follow-ups. In case there are some left, I will take care of opening issues. Would you mind if I added 'resolved' subfolder to your folder and moved resolved issues in there?

pcenov commented:
Hi @MisRob - please feel free to create a 'resolved' subfolder or any other folders as needed. :)

MisRob commented:
> Also, it appears that the LanguageSwitcherList component is not being displayed on the login page.

@AlexVelezLl fixed it in https://github.com/learningequality/kolibri-design-system/pull/612 so after I release the new KDS version next week, it will be resolved.

MisRob committed:
Merge branch 'release-v0.16.x' into rebrand

MisRob commented:
Summary of next steps on this PR:

- [x] @MisRob to release and install the new KDS with the fix for the language switcher list not being displayed on the login page, and select and modal issues
- [x] @MisRob to resolve https://drive.google.com/drive/u/1/folders/1C9jqrXs0h3ZYVCEpwyq5I7NAMdbBkEtO
- [x] @LianaHarris360 to resolve https://drive.google.com/drive/u/1/folders/13n0OWwzRR62eSI87Brtl2yQr_51Q2-jw
- [ ] @rtibbles to introduce the new logo

MisRob committed:
Fix position of buttons within search box

MisRob committed:
Remove Chrome cancel button from search inputs

We have our own cancel button implementation.
This fixes two cancel buttons being shown in Chrome.

MisRob committed:
Fix search chip contrast

MisRob committed:
Remove select override styles

Fixes contrast issues.

MisRob committed:
Remove select override styles

Fixes contrast issues.

MisRob commented:
@LianaHarris360 fixed the modal and select bugs found by @pcenov here https://github.com/learningequality/kolibri-design-system/pull/616 and it will be available after I create a new release and install it here.

MisRob committed:
Install KDS v4.2.0

MisRob commented:
All feedback should now be resolved

pcenov reviewed (approved):
Thanks @MisRob and @LianaHarris360 - I confirm that the reported issues are fixed now, and I didn't spot any new ones!

radinamatic reviewed (approved):
Manual QA passes, great job!!! 👏🏽 💯 :shipit:

MisRob commented:
Great work everybody :)!

akolson commented on kolibri/plugins/coach/assets/src/views/common/QuizStatus.vue:
```diff
@@ -312,8 +312,8 @@
-          'background-color': this.$themePalette.red.v_700,
-          ':hover': { 'background-color': this.$themePalette.red.v_900 },
+          'background-color': this.$themePalette.red.v_1100,
+          ':hover': { 'background-color': '#A81700' },
```
Is there a particular reason a hardcoded color code is used here?

MisRob replied:
Ah yes, the reason was that we can't use lighter red than `palette.red.v_1100` to ensure sufficient contrast, and at the same time we don't have darker red in the palette available. I discussed this with designers and they recommended to use this hardcoded color for this particular occasion (with that we may want to add it to the palette if we need it more times). I think in any case I should add a comment because it doesn't follow common pattern.

rtibbles replied:
Perhaps the way to do it would be to use a snippet like this? https://css-tricks.com/snippets/javascript/lighten-darken-color/ and then we programmatically darken/lighten a specific theme palette colour rather than hard coding the result?

MisRob replied:
I experimented with it and it works well, great idea. I imagine it as a KDS utility for which I'd like to add a bit of guidance on the Colors page. Designers had some thinking in progress about how to allow for colors outside of the palette so I will confirm with them at first.

MisRob replied:
I got confirmation from designers and we will do this, but I will work on it as follow-up since we need to clarify some things.

akolson reviewed (approved):
Hi @MisRob! Great work on this whale PR :). The changes look good to me! Thanks. 

I just left a non-blocking comment but thats about it from me!

MisRob committed:
Add a comment on usage of a hardcoded color

MisRob committed:
Merge branch 'release-v0.16.x' into rebrand
